src/Response.php

```diff
@@ -33,7 +36,7 @@ public function __construct(
         array $headers,
         InputStream $body,
         Request $request,
-        ConnectionInfo $connectionInfo,
+        Connection $connection,
```

Comment:
Response shouldn't have a reference to the connection itself.

Reply:
I suppose not… I was hoping to just drop `ConnectionInfo`, but maybe I'll have to keep it after all. Does it make sense to pass a `Connection` to `NetworkInterceptor` too?

Reply:
`NetworkInterceptor` can receive `Connection`, that's fine. We'll have to keep `ConnectionInfo` for `Response` or remove support for it in `Response` and if you want to access it, you have to implement a `NetworkInterceptor`.

Reply:
I think I like the second option, since 99.9% of the time you don't care about that information.